File: src/Java/myservicebus/src/main/java/com/myservicebus/serialization/NServiceBusJsonSerializerFactory.java
package com.myservicebus.serialization;

public final class NServiceBusJsonSerializerFactory implements SerializerFactory {
    @Override
    public String getContentType() {
        return DefaultInboundMessageResolver.RAW_JSON_CONTENT_TYPE;
    }

    @Override
    public MessageSerializer createSerializer() {
        return new NServiceBusJsonMessageSerializer();
    }

    @Override
    public MessageDeserializer createDeserializer() {
        return new NServiceBusJsonMessageDeserializer();
    }
}

File: src/MyServiceBus/Serialization/NServiceBusJsonSerializerFactory.cs
using System.Text.Json;

namespace MyServiceBus.Serialization;

public sealed class NServiceBusJsonSerializerFactory : ISerializerFactory
{
    private readonly JsonSerializerOptions _jsonSerializerOptions;

    public NServiceBusJsonSerializerFactory()
        : this(JsonSerializationDefaults.CreateNServiceBusOptions())
    {
    }

    public NServiceBusJsonSerializerFactory(JsonSerializerOptions jsonSerializerOptions)
    {
        ArgumentNullException.ThrowIfNull(jsonSerializerOptions);
        _jsonSerializerOptions = jsonSerializerOptions;
    }

    public string ContentType => InboundMessageResolver.RawJsonContentType;

    public IMessageSerializer CreateSerializer() => new NServiceBusJsonMessageSerializer(_jsonSerializerOptions);

    public IMessageDeserializer CreateDeserializer() => new NServiceBusJsonMessageDeserializer(_jsonSerializerOptions);
}
